internal/argocd/revision_test.go

```diff
@@ -99,39 +275,48 @@ func TestGetDesiredRevision(t *testing.T) {
 					},
 				},
 			},
-			freight: kargoapi.FreightReference{
-				Origin: testOrigin,
-				Commits: []kargoapi.GitCommit{
-					{
-						RepoURL:           "https://github.com/universe/42",
-						HealthCheckCommit: "fake-revision",
-						ID:                "bad-revision",
+			freightHistory: kargoapi.FreightHistory{
+				&kargoapi.FreightCollection{
+					Freight: map[string]kargoapi.FreightReference{
+						testOrigin.String(): {
+							Origin: testOrigin,
+							Commits: []kargoapi.GitCommit{
+								{
+									RepoURL:           "https://github.com/universe/42",
+									HealthCheckCommit: "fake-revision",
+									ID:                "bad-revision",
+								},
+							},
+						},
 					},
 				},
 			},
-			want: "fake-revision",
+			want: []string{"fake-revision"},
 		},
 	}
 
 	for _, testCase := range testCases {
 		t.Run(testCase.name, func(t *testing.T) {
-			stage := &kargoapi.Stage{
-				Spec: kargoapi.StageSpec{
-					PromotionMechanisms: &kargoapi.PromotionMechanisms{
-						ArgoCDAppUpdates: []kargoapi.ArgoCDAppUpdate{{
-							Origin: &testOrigin,
-						}},
-					},
-				},
-				Status: kargoapi.StageStatus{
-					FreightHistory: kargoapi.FreightHistory{{
-						Freight: map[string]kargoapi.FreightReference{
-							testOrigin.String(): testCase.freight,
+
+			stage := testCase.stage
+			if stage == nil {
+				stage = &kargoapi.Stage{
+					Spec: kargoapi.StageSpec{
+						PromotionMechanisms: &kargoapi.PromotionMechanisms{
+							ArgoCDAppUpdates: []kargoapi.ArgoCDAppUpdate{
+								{
+									Origin: &testOrigin,
+								},
+							},
 						},
-					}},
-				},
+					},
+					Status: kargoapi.StageStatus{
+						FreightHistory: testCase.freightHistory,
+					},
```

Comment:
There is no compelling reason to set this. There's actually a comment in `GetDesiredRevisions` about Freight _not_ being obtained from the status and being obtained instead from the last argument to the function because that grants the flexibility to use that function in the context of a Promotion _or_ a health check of a Stage with its current Freight.

Similarly, all the test cases that explicitly build out a State status with Freight history do not need to.

Reply:
Although I see this was a flaw in the original test as well.